marlowe-playground-client/src/Pretty.purs

```diff
@@ -41,3 +43,27 @@ renderPrettyPayee :: forall p i. MetaData -> Payee -> Array (HTML p i)
 renderPrettyPayee metadata (Account owner2) = [ text "account of ", renderPrettyParty metadata owner2 ]
 
 renderPrettyPayee metadata (Party dest) = [ text "party ", renderPrettyParty metadata dest ]
+
+showBigIntegerAsCurrency :: BigInteger -> Int -> String
+showBigIntegerAsCurrency number numDecimals = fromCharArray numberStr
+  where
+  absValStr = replicate (numDecimals + 1) '0' <> toCharArray (show (if number < zero then -number else number))
+
+  numDigits = length absValStr
+
+  numDigitsBeforeSeparator = numDigits - numDecimals
+
+  prefixStr = if number < zero then [ '-' ] else []
+
+  digitsNoZeros = dropWhile (\x -> x == '0') $ take numDigitsBeforeSeparator absValStr
+
+  digitsBeforeSeparator = if digitsNoZeros == [] then [ '0' ] else digitsNoZeros
+
+  digitsAfterSeparator = take numDecimals $ drop numDigitsBeforeSeparator (concat [ absValStr, replicate numDecimals '0' ])
+
+  numberStr = concat ([ prefixStr, digitsBeforeSeparator ] <> if digitsAfterSeparator /= [] then [ [ '.' ], digitsAfterSeparator ] else [])
+
```

Comment:
I wonder if we can reutilize this code here: https://github.com/input-output-hk/plutus/blob/master/marlowe-dashboard-client/src/Humanize.purs#L87

Reply:
I think the code you linked does something very different, and also I think the code you linked has the limitation that it can only format ada and dollars, so it doesn't work in the general case. This code just takes an integer and prints the result of dividing it by (10^numDecimals) with the proper formatting

Reply:
Ah, you meant this `toAda lovelace = (toNumber lovelace) / 1000000.0`, yes, pretty much, I guess I could do the same here, but not reuse it, because this one is generic

Reply:
well, I'd try to have a generic one and share it between the dashboard and playground, because entering and displaying numeric values with a format is something that we'd like to keep consistently across the apps.

Reply:
You guys probably want to choose yourselves a formatter solution as part of the redesign and then you can just replace this, I didn't mean to come up with the definitive solution. Then again I think we probably won't be able to push the formatter that is used for the input as far as the rest of formatters, I just reused this one because it is still improvement over no formatter, and kind of does the trick, but one that supports thousand separators would be better